scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,128 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+    
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.filterwarnings("ignore:Casting complex")
+def test_gttrf_gttrs(dtype):
+
+    np.random.seed(42)
+    # use ?gttrf and ?gttrs to solve a random linear system for each dtype
```

Comment:
I'd bring this comment to the top of the function and elaborate. For instance, you also check that return values of ?gttrf define LU matrices, etc... Summarize what this function tests.